lib/electric_slide/call_queue.rb

```diff
@@ -241,6 +241,8 @@ def remove_agent(agent, extra_params = {})
     # Checks to see if any callers are waiting for an agent and attempts to connect them to
     # an available agent
     def check_for_connections
+      # Ensure there are no nil objects in the call queue before trying to connect
+      @queue.compact!
```

Comment:
> My only thought is that the call dies and becomes nil somewhere between it being queued and getting connection, yet the connection is happening before the call is removed.

The thing is that values in an array can't become `nil`. The call can end, the actor can die, the value can become useless, but the value itself can't magically change from `Adhearsion::Call` to `nil`.

I can only think of two ways that a `nil` object would be enqueued:

1. It enters the queue that way from the application - perhaps adding logging to the code that adds things to the queue may reveal the source?
2. There is a yet-to-be-discovered bug in ElectricSlide that puts a `nil` into the queue - perhaps this can happen when a call connection fails? Maybe a call variable gets cleared and a `nil` value is re-added to the queue for the next agent? Adding trace logging to the code that puts a call in the queue array could reveal this